core/store/src/db/refcount.rs

```diff
@@ -126,10 +125,10 @@ pub(crate) fn refcount_merge<'a>(
         rc += delta;
     }
 
-    match rc.cmp(&0) {
-        Ordering::Less => rc.to_le_bytes().to_vec(),
-        Ordering::Equal => Vec::new(),
-        Ordering::Greater => [payload.unwrap_or(b""), &rc.to_le_bytes()].concat(),
+    if rc <= 0 {
+        Vec::new()
+    } else {
+        [payload.unwrap_or(b""), &rc.to_le_bytes()].concat()
```

Comment:
actual fix

Reply:
Please update the comment and add a TODO(resharding) with a short description that we should actually preserve negative refcounts but don't because of resharding.

Reply:
done in https://github.com/near/nearcore/pull/12784/commits/63c60d6c0e4c94017e0e97e0fc98e6ce38455c03